### DISCUSSION TITLE
Visual DL

### DISCUSSION THREAD

**Discussion by Baxulio** [PaddlePaddle/FastDeploy#1414] [Q&A] [answered]
Hello dears, 

I have some questions, can anyone please answer to them:

**1st question** 
I pulled and run docker image with the following command: 
```
docker run -it --net="host" --name fd_serving -v /:/serving registry.baidubce.com/paddlepaddle/fastdeploy:1.0.3-cpu-only-21.10
```
then did
```
visualdl --host 0.0.0.0 --port 8080
```
it runs, but I can't access to VisualDL from browser. What may be the cause?  My OS: Windows 11.

**2nd question** 
Does Visual DL or FastDeploy serving support pipelining different models. In other words does it support combining several models with each other? 

**3rd question** 
Does FastDeploy support other PaddlePaddle models in Puddle hub except for those in this repo. I'm more interested in vehicle recognition?

I appreciate all the answers from your side, 
Thank you in advance!

**Comment by jiangjiajun** [comment]
@rainyfly  Could you help to answer the 1st and 2rd questions?

And for the 3rd question. There's plan to support more models in FastDeploy,  you can feedback your model reqeust through the issue, and we will evaluate and decide whether to support it.

Also we also welcome everyone to integrate more models for FastDeploy, and we can provide support. If you are interested, you can refer this document https://github.com/PaddlePaddle/FastDeploy/blob/develop/docs/en/faq/develop_a_new_model.md

**Reply by rainyfly** [reply]
For 1st question, Could you please tell us which ip address you visited in your browser. The ip address should be the addr which starts VisualDL. Maybe the ip address you visited is wrong? For example, you launch VisualDL in a remote server, but your visited localhost or 0.0.0.0 in your local machine, which is the regular wrong operation.
For 2st question, fastdeployserver does support to pipeline multiple models. As you can see the serving examples in FastDeploy, the pipeline model is of type "ensemble", which consists of serveral models. For example, This is the YOLOv5 serving configuration, https://github.com/PaddlePaddle/FastDeploy/blob/develop/examples/vision/detection/yolov5/serving/models/yolov5/config.pbtxt , which is consist of serveral basic units, i.e. preprocess, runtime and postprocess . You can write a config.pbtxt like this, pipeline several models into one for serving.

**Reply by Baxulio** [reply]
Thank you so much for the explanation! 

I'm doing everything on my machine. I don't use remote servers. And trying to access VisualDL by
```
http://127.0.0.1:8080/
```

Maybe this is because of windows. Can you tell me please, are the commands right in general, or I missed something?

Thank you!

**Comment by rainyfly** [comment, marked as answer]
Oh, it is confusing. I noticed you have add --net=host in command "docker run -it --net="host" --name fd_serving -v /:/serving registry.baidubce.com/paddlepaddle/fastdeploy:1.0.3-cpu-only-21.10", so your docker environment does share the same network with your host. Can you please check that you can visit the network in your docker environment. For example, you can use ' python -m http.server 8080 ' to launch a http server in your docker, then you can try to visit the http server to check the network.

**Comment by Baxulio** [comment]
Yes, I think this is an issue related to OS Windows. I tried on Ubuntu, works like a charm. also can see the ensemble functionality. Thank you a lot! Have a great day & coding :)
